From d8575412eb50c8758f64e9bb37363c974a0a36ac Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Thu, 1 Dec 2016 20:51:46 +0000
Subject: [PATCH] [CONFIGURATION-641] Added a warning to FileBased Javadocs.

The methods defined by this interface should not be called directly
by client code, but only via a FileHandler object.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/trunk@1772264 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/commons/configuration2/io/FileBased.java | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/main/java/org/apache/commons/configuration2/io/FileBased.java b/src/main/java/org/apache/commons/configuration2/io/FileBased.java
index 6541cc4c8c..229c9f22c5 100644
--- a/src/main/java/org/apache/commons/configuration2/io/FileBased.java
+++ b/src/main/java/org/apache/commons/configuration2/io/FileBased.java
@@ -34,6 +34,15 @@
  * these methods it is possible to implement other methods which operate on
  * files, file names, URLs, etc.
  * </p>
+ * <p>
+ * <strong>Note that the methods defined by this interface are not intended to
+ * be called directly by client code!</strong> Rather, they are used internally
+ * when doing I/O operations with a {@link FileHandler}. A {@code FileHandler}
+ * supports additional functionality (e.g. it evaluates some additional
+ * interfaces the {@code FileBased} object may implement); this functionality
+ * is not available on a direct method invocation, so this may lead to
+ * unpredictable results.
+ * </p>
  *
  * @version $Id$
  */
@@ -41,6 +50,8 @@ public interface FileBased
 {
     /**
      * Reads the content of this object from the given reader.
+     * <strong>Client code should not call this method directly, but use a
+     * {@code FileHandler} for reading data.</strong>
      *
      * @param in the reader
      * @throws IOException if an I/O error occurs
@@ -51,6 +62,8 @@ public interface FileBased
 
     /**
      * Writes the content of this object to the given writer.
+     * <strong>Client code should not call this method directly, but use a
+     * {@code FileHandler} for writing data.</strong>
      *
      * @param out the writer
      * @throws IOException if an I/O error occurs
